bob-common/src/data.rs

```diff
@@ -92,26 +94,44 @@ pub type VDiskId = u32;
 
 #[derive(Clone)]
 pub struct BobData {
-    inner: Vec<u8>,
+    inner: Bytes,
     meta: BobMeta,
 }
 
 impl BobData {
-    pub fn new(inner: Vec<u8>, meta: BobMeta) -> Self {
+    const TIMESTAMP_LEN: usize = 8;
+    
+    pub fn new(inner: Bytes, meta: BobMeta) -> Self {
         BobData { inner, meta }
     }
 
     pub fn inner(&self) -> &[u8] {
         &self.inner
     }
 
-    pub fn into_inner(self) -> Vec<u8> {
+    pub fn into_inner(self) -> Bytes {
         self.inner
     }
 
     pub fn meta(&self) -> &BobMeta {
         &self.meta
     }
+
+    pub fn to_vec(&self) -> Vec<u8> {
+        let mut result = self.meta.timestamp.to_be_bytes().to_vec();
+        result.extend_from_slice(&self.inner);
+        result
+    }
+
+    pub fn from_bytes(data: &[u8]) -> Result<BobData, Error> {
```

Comment:
There is only one place, from where this function is called. In that place vector is used as parameter. Thus it would be better to receive a vector here, create `Bytes` from that vector and then manipulate that Bytes struct. It has `split_to` function and can be dereferenced as array to call `u64::from_be_bytes`. This possibly help to avoid full data copying.